File: Lesson.02 - Design Patterns - Creational/FactoryMethod/src/main/java/com/github/ASE/FactoryMethod/HTML/DocumentGenerator.java
package com.github.ASE.FactoryMethod.HTML;

import com.github.ASE.FactoryMethod.Footer;
import com.github.ASE.FactoryMethod.Header;
import com.github.ASE.FactoryMethod.Paragraph;
import com.github.ASE.FactoryMethod.Table;

public class DocumentGenerator extends com.github.ASE.FactoryMethod.DocumentGenerator {

    @Override
    public Header createHeader() {
        return new Header() {
            @Override
            public String getContent() {
                return "<h1>Document Title</h1>";
            }
        };
    }

    @Override
    public Paragraph createParagraph() {
        return new Paragraph() {
            @Override
            public String getContent() {
                return "<p>This is an HTML paragraph.</p>";
            }
        };
    }

    @Override
    public Table createTable() {
        return new Table() {
            @Override
            public String getContent() {
                return "<table border=\"1\"><tr><th>Name</th><th>Age</th></tr><tr><td>John</td><td>30</td></tr></table>";
            }
        };
    }

    @Override
    public Footer createFooter() {
        return new Footer() {
            @Override
            public String getContent() {
                return "<footer>End of Document</footer>";
            }
        };
    }

}

File: Lesson.02 - Design Patterns - Creational/Python/FactoryMethod/html/document_generator.py
from document_generator import DocumentGenerator
from components.header import Header
from components.paragraph import Paragraph
from components.table import Table
from components.footer import Footer


class HTMLDocumentGenerator(DocumentGenerator):
    def create_header(self) -> Header:
        return HTMLHeader()

    def create_paragraph(self) -> Paragraph:
        return HTMLParagraph()

    def create_table(self) -> Table:
        return HTMLTable()

    def create_footer(self) -> Footer:
        return HTMLFooter()


class HTMLHeader(Header):
    def get_content(self) -> str:
        return "<h1>Document Title</h1>"


class HTMLParagraph(Paragraph):
    def get_content(self) -> str:
        return "<p>This is an HTML paragraph.</p>"


class HTMLTable(Table):
    def get_content(self) -> str:
        return '<table border="1"><tr><th>Name</th><th>Age</th></tr><tr><td>John</td><td>30</td></tr></table>'


class HTMLFooter(Footer):
    def get_content(self) -> str:
        return "<footer>End of Document</footer>"
